tla-types/src/test/scala/at/forsyte/apalache/tla/typecheck/etc/TestTypeCheckerTool.scala

```diff
@@ -25,38 +26,202 @@ class TestTypeCheckerTool extends FunSuite with BeforeAndAfterEach with EasyMock
   private var sourceStore: SourceStore = _
   private var annotationStore: AnnotationStore = _
   private var sanyImporter: SanyImporter = _
+  private var parser: Type1Parser = _
 
   override def beforeEach() {
     sourceStore = new SourceStore()
     annotationStore = createAnnotationStore()
     sanyImporter = new SanyImporter(sourceStore, annotationStore)
+    parser = DefaultType1Parser
   }
 
-  test("the tool runs") {
-    val text =
-      """---- MODULE Specb -----
-        |EXTENDS Integers
-        |VARIABLES
-        |   \* @type: Set([type: Str, val: Int]);
-        |   msgs
-        |
-        |\* @type: [type: Str, val: Int] => Bool;
-        |Send(m) ==
-        |  (msgs' = msgs \cup {m})
-        |
-        |A(x, y) == x + y
-        |
-        |Init ==
-        |  msgs = {}
-        |
-        |Next ==
-        |  \E i \in 1..10:
-        |    Send([type |-> "1a", val |-> i])
-        |================================
+  // We push plenty of TLA+ constructs in this spec, to make sure that:
```

Comment:
I wonder if it wouldn't be more maintainable to just read this from a file?

Reply:
Good point. So far I did want to mess with java class paths. But having this test in an actual file could improve the code.